test/specs/lib/reactComponentUtils-test.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react'
+import { getComponentName, areTypeNamesEqual } from 'src/lib/reactComponentUtils'
+import { createMockComponent } from 'test/utils'
+
+describe('getComponentName', () => {
+  it('returns null for invalid arguments', () => {
+    const invalidArgs = [undefined, null, '', {}]
+    invalidArgs.map(invalidArg => expect(getComponentName(invalidArg as any)).toBeFalsy())
+  })
+
+  it('returns the string component itself', () => {
+    const stringArgs = ['div', 'span', 'MyComponent']
+    stringArgs.map(stringArg => expect(getComponentName(stringArg)).toEqual(stringArg))
+  })
+
+  it('returns the function name', () => {
+    const MyComponent = createMockComponent('MyComponent')
+
+    function fooComponent() {
+      return null
+    }
+
+    expect(getComponentName(fooComponent)).toEqual('fooComponent')
+    expect(getComponentName(MyComponent)).toEqual('MyComponent')
+  })
+})
+
+describe('areTypeNamesEqual', () => {
+  it('returns false for invalid arguments', () => {
+    expect(areTypeNamesEqual(null, undefined)).toBeFalsy()
+    expect(areTypeNamesEqual(undefined, '')).toBeFalsy()
+    expect(areTypeNamesEqual('', '')).toBeFalsy()
+  })
+
+  describe('for primitive components', () => {
+    it('returns false for different types', () => {
+      expect(areTypeNamesEqual(React.createElement('p').type, 'div')).toBeFalsy()
```

Comment:
not sure why it is necessary to test two semantically absolutely equal cases. Would rather suggest to merge tests under describe blocks and simplify tests hierarchy (as these tests are fairly simple):

```js
describe('areTypeNamesEqual', () => {
  test('properly handles string component types', ...)
  test('properly handles functional component types', ...)
})
```
For me as a client it would absolutely obvious what does this **properly handes* implies by default given the name of the function - so, there is no need to introduce additional test cases to further describe this behavior for each set of component types.

Reply:
N.A. anymore since the implementation was drastically simplified following decision:
https://github.com/stardust-ui/react/pull/496#discussion_r238708722